src/roundtrip.cpp

```diff
@@ -3452,14 +3554,17 @@ static int run_in_uhdm_mode(int argc, const char **argv) {
     return -2;
   }
 
-  for (const vpiHandle &handle : restoredDesigns) {
-    const UHDM::design *const design =
-        (const UHDM::design *)((const uhdm_handle *)handle)->object;
-    if (!design->VpiElaborated()) {
-      UHDM::ElaboratorListener listener(&serializer, false);
-      listener.listenDesigns(restoredDesigns);
-    }
-  }
+  // Do NOT elaborate explicitly since multiple designs are incomplete and
+  // an elaboration process for them doesn't end up well.
+  //
+  // for (const vpiHandle &handle : restoredDesigns) {
+  //   const UHDM::design *const design =
+  //       (const UHDM::design *)((const uhdm_handle *)handle)->object;
+  //   if (!design->VpiElaborated()) {
+  //     UHDM::ElaboratorListener listener(&serializer, false);
```

Comment:
This should now be changed to UHDM::ElaboratorListener listener(&serializer, false, true);
if you need elaboration.